src/java/picard/vcf/GvcfMetricAccumulator.java

```diff
@@ -0,0 +1,56 @@
+package picard.vcf;
+
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import htsjdk.variant.variantcontext.VariantContext;
+import picard.util.DbSnpBitSetUtil;
+
+import java.util.List;
+
+/**
+ * Collects variants from single sample GVCF and generates metrics about them.
+ *
+ * @author farjoun
+ */
+public class GvcfMetricAccumulator extends CallingMetricAccumulator {
+    String sample = null;
+
+    public GvcfMetricAccumulator(final DbSnpBitSetUtil.DbSnpBitSets dbsnp) {
+        super(dbsnp);
+    }
+
+    @Override
+    public void accumulate(final VariantContext vc) {
+        if (sample == null) {
+            final List<String> samples = vc.getSampleNamesOrderedByName();
+            if (samples == null || samples.size() != 1) {
+                throw new IllegalArgumentException("Expected to have exactly 1 sample in a GVCF, found " + ((samples == null) ? "NULL" : samples.size()));
+            }
+            sample = samples.get(0);
+        }
+        final VariantContext subContext = vc.subContextFromSample(sample);
```

Comment:
to remove the non-ref allele, which is why this whole thing is needed to begin with.